### PR TITLE
Bug 1109 metrics fields

### LINKED ISSUE #1109: Metrics fields: an empty field causes problems, breaks data

### PR DESCRIPTION
**Pull request by ollimeier**
Fixes #1109 

I am not able to get and set the value before, but I guess it would require a more. For now, this is a fix with an info message and after that you can undo you change. What do you think @justvanrossum

### CODE REVIEW

**Comment by justvanrossum** [comment]
Quick thought: the initialValue should at least be optional. It makes no sense in the non-editing field types. But let me think a bit longer, I think we should just store the previous value so we can access it.

File: src/fontra/views/editor/panel-selection-info.js [hunk 1]
```diff
@@ -463,6 +482,7 @@ export default class SelectionInfoPanel extends Panel {
     }
 
     this.infoForm.onFieldChange = async (fieldItem, value, valueStream) => {
+      fieldItem.initialValue = value;
```
Comment (ollimeier) [inline src/fontra/views/editor/panel-selection-info.js:485]: @justvanrossum  This is the line of code, where I store the latest value.

Comment (justvanrossum) [reply src/fontra/views/editor/panel-selection-info.js:485]: That is not correct: we shouldn't use the field item that way.

**Comment by justvanrossum** [comment]
Also: we should never just pop up a dialog in such a case.

File: src/fontra/client/web-components/ui-form.js [hunk 2]
```diff
@@ -181,6 +183,11 @@ export class Form extends SimpleElement {
     };
     inputElement.onchange = (event) => {
       let value = parseFloat(inputElement.value);
+      if (isNaN(value)) {
```
Comment (justvanrossum) [inline src/fontra/client/web-components/ui-form.js:186]: Let's move this bit to just before the `this._lastValidFieldValues[fieldItem.key] = value;` line, that way it is more similar to the rotation case.

Comment (ollimeier) [reply src/fontra/client/web-components/ui-form.js:186]: Sorry, but I don't know where you want to move it.

Comment (ollimeier) [reply src/fontra/client/web-components/ui-form.js:186]: Isn't it pretty much the same? `if (isNaN(value)) {` is right after `let value = parseFloat(inputElement.value);` in both cases.

Comment (justvanrossum) [reply src/fontra/client/web-components/ui-form.js:186]: Rereading the code: never mind, the current place is best after all.

(I was hoping the code would look more similar to the rotation case, but I think it indeed needs to be before the "reportValidity" section.

Comment (ollimeier) [reply src/fontra/client/web-components/ui-form.js:186]: 'Code Design' ;)

**Pull request merged**